setup/steps/establish_superuser_connection.go

```diff
@@ -0,0 +1,152 @@
+			if selectedPg.Port == 0 {
+				var socketDirStr string
+				if state.Inputs.PGSetupConnSocketDir.Valid {
+					socketDirStr = " in " + state.Inputs.PGSetupConnSocketDir.String
+				}
+
+				return fmt.Errorf(
+					"no Postgres server found listening on %d%s",
+					state.Inputs.PGSetupConnPort.Int64,
+					socketDirStr,
+				)
+			}
+		} else {
+			if len(localPgs) == 0 {
+				return errors.New("failed to find a running local Postgres install")
+			} else if len(localPgs) == 1 {
+				selectedPg = localPgs[0]
+			} else {
+				var opts []string
+				for _, localPg := range localPgs {
+					opts = append(opts, fmt.Sprintf("port %d in socket dir %s", localPg.Port, localPg.SocketDir))
+				}
+				var selectedIdx int
+				err := survey.AskOne(&survey.Select{
+					Message: "Found several Postgres installations; please select one",
+					Options: opts,
+				}, &selectedIdx)
+				if err != nil {
+					return err
+				}
+				selectedPg = localPgs[selectedIdx]
+			}
+		}
+
+		var pgSuperuser string
+		if state.Inputs.Scripted {
+			if !state.Inputs.PGSetupConnUser.Valid {
+				return errors.New("no user specified for setup Postgres connection")
+			}
+			pgSuperuser = state.Inputs.PGSetupConnUser.String
+		} else {
+			err = survey.AskOne(&survey.Select{
+				Message: "Select Postgres superuser to connect as for initial setup:",
+				Help:    "We will create a separate, restricted monitoring user for the collector later",
+				Options: []string{"postgres", "another user..."},
+			}, &pgSuperuser)
+			if err != nil {
+				return err
+			}
+			if pgSuperuser != "postgres" {
+				err = survey.AskOne(&survey.Input{
+					Message: "Enter Postgres superuser to connect as for initial setup:",
+					Help:    "We will create a separate, restricted monitoring user for the collector later",
+				}, &pgSuperuser, survey.WithValidator(survey.Required))
+				if err != nil {
+					return err
+				}
+			}
+		}
+
+		state.QueryRunner = query.NewRunner(pgSuperuser, selectedPg.SocketDir, selectedPg.Port)
+
+		return nil
+	},
+}
+
+type LocalPostgres struct {
+	SocketDir string
+	LocalAddr string
+	Port      int
+}
+
+var pgsqlDomainSocketPortRe = regexp.MustCompile("\\d+$")
+
+func getSocketDirMatches(dir string) ([]LocalPostgres, error) {
+	var result []LocalPostgres
+	// technically this should be a filepath.Join, but Unix-domain sockets do not work
+	// on windows anyway
+	globPattern := fmt.Sprintf("%s/.s.PGSQL.*", dir)
+	matches, err := filepath.Glob(globPattern)
+	if err != nil {
+		return nil, err
+	}
+	for _, match := range matches {
+		portStr := pgsqlDomainSocketPortRe.FindString(match)
+		if portStr == "" {
+			continue
+		}
+		port, err := strconv.Atoi(portStr)
+		if err != nil {
+			return nil, err
+		}
+		result = append(result, LocalPostgres{SocketDir: dir, Port: port})
+	}
+	return result, nil
+}
+
+func discoverLocalPostgres() ([]LocalPostgres, error) {
+	// TODO: find tcp sockets if no unix sockets?
+	// TODO: confirm these are live by checking pids?
```

Comment:
We can remove these TODOs as discussed - maybe clarify the function name by calling it `discoverLocalPostgresFromUnixDomainSockets` (or similar)